node/src/components/consensus/protocols/zug.rs

```diff
@@ -941,8 +956,14 @@ impl<C: Context + 'static> Zug<C> {
             signed_messages,
             evidence,
             instance_id,
+            sync_id,
         } = sync_response;
 
+        // We have not asked for any sync response:
+        if self.sent_sync_requests.try_remove_id(sync_id) {
```

Comment:
If we change `try_remove_id` to match the comment (see below), we should also reverse the logic here: `if !self.sent_sync_requests.try_remove_id(sync_id)`.

Reply:
I don't really like inversed logic, that's why I did it that way…